driver_test.go

```
@@ -0,0 +1,176 @@
+package neo4j
+
+import (
+	"testing"
+
+	"github.com/cloudprivacylabs/lsa/layers/cmd/cmdutil"
+	"github.com/cloudprivacylabs/lsa/pkg/ls"
+	"github.com/cloudprivacylabs/opencypher"
+	"github.com/cloudprivacylabs/opencypher/graph"
+	"github.com/neo4j/neo4j-go-driver/v4/neo4j"
+)
+
+var seq int = -1
+
+var expected = []struct {
+	sources       []neo4jNode
+	targets       []neo4jNode
+	edges         []neo4jEdge
+	expectedGraph graph.Graph
+}{
+	{
+		sources: []neo4jNode{
+			{Id: 60, Labels: []string{ls.DocumentNodeTerm}, Props: map[string]interface{}{shortEntitySchemaTerm: "Schema for a city"}},
+		},
+		targets: []neo4jNode{
+			{Id: 61, Labels: []string{ls.DocumentNodeTerm}, Props: map[string]interface{}{"value": "San Francisco"}},
+			{Id: 62, Labels: []string{ls.DocumentNodeTerm}, Props: map[string]interface{}{"value": "Denver"}},
+		},
+		edges: []neo4jEdge{
+			{Id: 28, Props: map[string]interface{}{"type": "city"}, StartId: 60, EndId: 61},
+			{Id: 29, Props: map[string]interface{}{"type": "city"}, StartId: 60, EndId: 62},
+		},
+	},
+}
+
+func mockFindNeighbor(tx neo4j.Transaction, ids []int64) ([]neo4jNode, []neo4jNode, []neo4jEdge, error) {
+	seq++
+	if seq < len(expected) {
```

Don't check, let it panic.